libs/blocks/merch/merch.js

```diff
@@ -24,86 +18,97 @@ const GEO_MAPPINGS = {
   no: 'nb-NO',
 };
 
-const PLACEHOLDER_TYPE_MAPPINGS = {
-  optical: 'priceOptical',
-  strikethrough: 'priceStrikethrough',
-};
-
-let initialized = false;
-
-function omitNullValues(target) {
+const omitNullValues = (target) => {
   if (target != null) {
     Object.entries(target).forEach(([key, value]) => {
       if (value == null) delete target[key];
     });
   }
   return target;
-}
+};
 
-const getTacocatEnv = (envName, locale) => {
-  const wcsLocale = GEO_MAPPINGS[locale.prefix] ?? locale.ietf;
-  // eslint-disable-next-line prefer-const
-  let [language, country = 'US'] = wcsLocale.split('-', 2);
+window.tacocat = window.tacocat || {};
+
+export const imsCountryPromise = () => new Promise((resolve) => {
+  let count = 0;
+  const check = setInterval(() => {
+    count += 1;
+    if (window.adobeIMS) {
+      clearInterval(check);
+      if (window.adobeIMS.isSignedInUser()) {
+        window.adobeIMS
+          .getProfile()
+          .then(({ countryCode }) => resolve(countryCode))
+          .catch(() => resolve());
+      } else {
+        resolve();
+      }
+    } else if (count > 25) {
```

Comment:
can we somehow cover this use case? maybe stubbing isSignedInUser, why 25 btw :) ?

Reply:
25 * 200 = 5000 second before failing the promise.

Reply:
could you add it as a comment there @yesil ?

Reply:
@3ch023 any unnecessary byte is bad for performance. reading the code it is obvious.
And this could should vanish as soon as milo ims support is ready.

cc: @chrischrischris